cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
+            lookups = []
+            for field in self.schema[obj].values():
+                if field["type"] == "reference":
+                    lookups.append(field["name"])
+                else:
+                    fields.append(field["name"])
+            self.mapping[key]["fields"] = OrderedDict()
+            if fields:
+                if "Id" not in fields:
+                    fields.append("Id")
+                fields.sort(key=field_sort)
+                for field in fields:
+                    self.mapping[key]["fields"][field] = (
+                        field.lower() if field != "Id" else "sf_id"
+                    )
+            if lookups:
+                lookups.sort(key=field_sort)
+                self.mapping[key]["lookups"] = OrderedDict()
+                for field in lookups:
+                    self.mapping[key]["lookups"][field] = {
+                        "table": self.schema[obj][field]["referenceTo"][0].lower()
+                    }
+                    if len(self.schema[obj][field]["referenceTo"]) > 1:
+                        self.logger.warning(
+                            "Field {}.{} is a polymorphic lookup, which is not supported".format(
+                                obj, field
+                            )
+                        )
+
+    def _run_task(self):
+        self._collect_objects()
+        self._build_schema()
+        self._build_mapping()
+        with open(self.options["path"], "w") as f:
+            yaml.safe_dump(self.mapping, f)
+
+    def _split_dependencies(self, objs, dependencies):
+        stack = []
+        objs_remaining = objs.copy()
+
+        # The structure of `dependencies` is:
+        # key = object, value = set of objects it references.
+
+        # Iterate through our list of objects
+        # For each object, if it is not dependent on any other objects, place it at the end of the stack.
+        # Once an object is placed in the stack, remove dependencies to it (they're satisfied)
+        while objs_remaining:
+            objs_without_deps = list(
+                filter(
+                    lambda obj: obj not in dependencies or not dependencies[obj],
+                    objs_remaining,
+                )
+            )
+
+            if not objs_without_deps:
+                self.logger.error(
+                    "Unable to complete mapping; the schema contains reference cycles or unresolved dependencies."
+                )
+                self.logger.info("Mapped objects: {}".format(", ".join(stack)))
+                self.logger.info("Remaining objects:")
+                for obj in objs_remaining:
+                    self.logger.info(
+                        "{} (depends on {})".format(
+                            obj,
+                            ", ".join(dependencies[obj]) if obj in dependencies else "",
+                        )
+                    )
+                raise BulkDataException("Cannot complete mapping")
```

Comment:
We should make BulkDataException subclass CumulusCIFailure instead of CumlusCIException. That way the CLI will print out the error message without a traceback.